File: Python/Labs/Module 3/Lab 1/Q1 - Maximize Number Of Subsequences In A String/main.py
text=input()
pattern=input()
res=cnt1=cnt2=0
for c in text:
    if c==pattern[1]:
        res+=cnt1
        cnt2+=1
    if c==pattern[0]:
        cnt1+=1
print(res+max(cnt1,cnt2))
                                                                                                                            

File: C++/Labs/Module 3/Lab 1/Q1 - Maximize Number Of Subsequences In A String/main.cpp
#include <iostream>
#include <string>
using namespace std;

int main(){
    string text,pattern;
    cin>>text>>pattern;
    int res=0,c1=0,c2=0;
    for(char c:text){
        if(c==pattern[1]){
            res+=c1;
            c2++;
        }
        if(c==pattern[0]){
            c1++;
        }
    }
    cout<<res+max(c1,c2);
    return 0;
}
                                                                                                                            